### PR TITLE
feat(python): suggest str.to_datetime instead of apply and stdlib strptime

### PR DESCRIPTION
**Pull request by MarcoGorelli**

### CODE REVIEW

File: py-polars/polars/utils/udfs.py [hunk 1]
```diff
@@ -685,6 +687,65 @@ def _rewrite_methods(
...
+            expr_name = f"str.to_datetime(format='{fmt}')"
+            synthetic_call = inst1._replace(
+                opname="POLARS_EXPRESSION",
+                argval=expr_name,
+                argrepr=expr_name,
+                offset=inst3.offset,
+            )
+            # POLARS_EXPRESSION is mapped as a unary op, so switch instruction order
+            operand = inst3._replace(offset=inst1.offset)
+            updated_instructions.extend((operand, synthetic_call))
```
Comment (MarcoGorelli) [inline py-polars/polars/utils/udfs.py:745]: not totally sure about this part (tests passing notwithstanding), gonna check it more carefully

**Comment by alexander-beedie** [comment]
FYI: I just [streamlined](https://github.com/pola-rs/polars/pull/10265) `BINARY_SUBSCR` matching; wonder if we can streamline matching on `module.class.func` /  `class.func` too...🤔

**Comment by MarcoGorelli** [comment]
>  wonder if we can streamline matching on "module.class.func" / "class.func" too

yup, good idea! have given this a go

**Comment by alexander-beedie** [comment]
> yup, good idea! have given this a go

And I have finally got around to reviewing it!
Verdict: sneaky - I like it :)

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**